CMake/version.cmake

```diff
@@ -1,12 +1,13 @@
 set(tmp)
 find_package(Git QUIET)
 if(GIT_FOUND)
-    execute_process(COMMAND ${GIT_EXECUTABLE}
-        --git-dir=${CMAKE_SOURCE_DIR}/.git describe --tags
-        OUTPUT_VARIABLE tmp OUTPUT_STRIP_TRAILING_WHITESPACE)
+  execute_process(COMMAND ${GIT_EXECUTABLE}
+      --git-dir=${CMAKE_SOURCE_DIR}/.git describe --tags
+      OUTPUT_VARIABLE tmp OUTPUT_STRIP_TRAILING_WHITESPACE)
 endif()
 if(NOT tmp)
-    set(tmp "v0.0.0")
```

Comment:
set this to v4.0.0 and add a comment that this is needed for spack builds which use tarballs rather than git